File: ProductManager/src/main/java/com/spring/productmanager/entities/Product.java
package com.spring.productmanager.entities;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.Instant;

@Getter
@Setter
@Entity
@Table(name = "product")
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "_id_product")
    private Long idProduct;

    @Column(name = "_product_code")
    private String productCode;

    @Column(name = "_product_name")
    private String productName;

    @ManyToOne
    @JoinColumn(name = "_id_brand")
    private Brand idBrand;

    @ManyToOne
    @JoinColumn(name = "_id_category")
    private Category idCategory;

    @ManyToOne
    @JoinColumn(name = "_id_sole")
    private Sole idSole;

    @ManyToOne
    @JoinColumn(name = "_id_material")
    private Material idMaterial;

    @Column(name = "_image")
    private String images;

    @Column(name = "_date_created")
    private Instant dateCreated;

    @Column(name = "_date_updated")
    private Instant dateUpdated;

}

File: ProductManagerApp/src/app/class/product.ts
import { Brand } from './brand';
import { Category } from './category';
import { Material } from './material';
import { Sole } from './sole';

export class Product {
  idProduct: number = 0;
  productCode: string = '';
  productName: string = '';
  idBrand: Brand = new Brand();
  idCategory: Category = new Category();
  idSole: Sole = new Sole();
  idMaterial: Material = new Material();
  images: string = '';
  dateCreated: Date = new Date();
  dateUpdated: Date = new Date();
}
